orthoinference/src/main/java/org/reactome/orthoinference/StableIdentifierGenerator.java

```diff
@@ -40,14 +40,14 @@ public GKInstance generateOrthologousStableId(GKInstance inferredInst, GKInstanc
         // For now, Human is hard-coded as the source species, so we replace the stableIdentifier source species based on that assumption
         String sourceIdentifier = (String) stableIdentifierInst.getAttributeValue(identifier);
         String targetIdentifier = sourceIdentifier.replace("HSA", speciesAbbreviation);
-
         // Paralogs will have the same base stable identifier, but we want to denote when that happens.
         // We pull the value from `seenOrthoIds`, increment it and then add it to the stable identifier name (eg: R-MMU-123456-2)
         int paralogCount = Optional.ofNullable(seenOrthoIds.get(targetIdentifier)).orElse(0) + 1;
+        String targetIdentifierUnmodified = targetIdentifier;
         if (paralogCount > 1) {
             targetIdentifier += "-" + paralogCount;
         }
-        seenOrthoIds.put(targetIdentifier, paralogCount);
+        seenOrthoIds.put(targetIdentifierUnmodified, paralogCount);
```

Comment:
Maybe this line can be moved to just after the initialization of the `targetIdentifier` on line 42.  That way, it will get stored before it is possibly modified by the paralogCount logic and you won't need the `targetIdentifierUnmodified` temp variable.  One less thing to understand and keep track of.

Reply:
Good point, I'll make that change now